### ISSUE TITLE
Need solution for asyncio SSL/TLS connections

### ISSUE THREAD

**Issue by RuofengX** [micropython/micropython#8647]
I am running `esp32c3-usb-20220117-v1.18.bin`

## TL;DR 

How to open a SSL or TLS connection using uasyncio?

------

I am tryign to port a CPython project onto Micropython.   

In CPython asyncio, we could open a SSL connection easily by using `ssl` argument in `asyncio.open_connection()` method.   
When I am trying to use `uasyncio.open_connection(..., ssl=True)`, it raises an exception saying that there's no ssl argument. We can confirm it on the doc page [here](https://docs.micropython.org/en/latest/library/uasyncio.html#uasyncio.open_connection).

Then I try to use ssl module to warp the socket in `StreamWriter.get_extra_info("socket")`(I never test this, I don't know whether this would work,  just have a try), but I cannot get the "raw" socket of this connection, because function `uasyncio.Stream.get_extra_info` only accept "peername" as a param, no support for "socket". We can check it [here](https://docs.micropython.org/en/latest/library/uasyncio.html#uasyncio.Stream.get_extra_info).

If there's a method to open an SSL/TLS connection by using uasyncio? **I really want to use asyncio because I think that is the best part of python language.**

**Comment by RuofengX** [comment]
Here is the code https://github.com/RuofengX/uPyxy/blob/c3772838be208056282e01ed0d3f8da0c8fa4de8/main/lib/client.py#L103

**Comment by peterhinch** [comment]
You might like to look at [mqtt_as](https://github.com/peterhinch/micropython-mqtt) which uses nonblocking sockets and works over TLS.

**Comment by RuofengX** [comment]
> I implemented a workaround for my project [here](https://github.com/hyx0329/micropython-uicefox/blob/b9af3acfe80587b765d39eb9803605f33561138b/uicefox.py#L7). It's basically a custom connection maker based on the original one.

That's awesome! I will check it right now😍

**Comment by stephanelsmith** [comment]
I ran into this last week and also implemented my own asyncio friendly implementation.  My version is optimized for my application, but asyncio ssl open_connection is surely something that would be apprecaited by future travelers.

I will also agree asyncio is my favorite part of python language!  Game changer for micro-controller dev.

**Comment by dpgeorge** [comment]
Asyncio SSL server and client are now supported.
